frontend/lib/identification/identification_qr_content_parser.dart

```diff
@@ -0,0 +1,122 @@
+                "Der Inhalt des eingescannten Codes kann nicht verstanden "
+                "werden. Vermutlich handelt es sich um einen QR Code, "
+                "der nicht für die Ehrenamtskarte App generiert wurde.");
+      }
+
+      final fullName = cardActivateModel.fullName;
+      if (fullName == null) {
+        return createFieldMissingError("fullName", "fullNameMissing");
+      }
+      final randomBytes = cardActivateModel.randomBytes;
+      if (randomBytes == null) {
+        return createFieldMissingError("randomBytes", "randomBytesMissing");
+      }
+      final unixInt64ExpirationDate = cardActivateModel.expirationDate;
+      int unixExpirationDate;
+      if (unixInt64ExpirationDate != null) {
+        try {
+          unixExpirationDate = unixInt64ExpirationDate.toInt();
+          if (unixExpirationDate == null) {
+            throw Exception(["unixExpirationDate was null"]);
+          }
+        } on Exception catch (e) {
+          return QRCodeParseResult(
+              hasError: true,
+              internalErrorMessage: "Failed to parse unixExpirationDate, "
+                  "Message: ${e.toString()}",
+              userErrorMessage: "Beim Verarbeiten des Ablaufdatums ist ein "
+                  "unerwarteter Fehler aufgetreten.");
+        }
+      }
+      if (cardActivateModel.cardType == CardActivateModel_CardType.STANDARD &&
+          unixExpirationDate == null) {
+        return QRCodeParseResult(
+            hasError: true,
+            internalErrorMessage: "Got card type "
+                "${cardActivateModel.cardType.toString()} but "
+                "unixExpirationDate was not set",
+            userErrorMessage: "Die eingescannte Karte enthält kein Ablauf-"
+                "datum, obwohl dies für die blaue Ehrenamtskarte erforderlich"
+                " ist. Vermutlich ist beim Erstellen der "
+                "digitalen Ehrenamtskarte ein Fehler passiert.");
+      }
+      final cardType = cardActivateModel.cardType.toString();
+      final regionId = cardActivateModel.region;
+      String base32TotpSecret;
+      try {
+        base32TotpSecret = base32.encode(cardActivateModel.totpSecret);
+      } on Exception catch (e) {
+        return QRCodeParseResult(
+            hasError: true,
+            internalErrorMessage: "Failed to parse base32TotpSecret, "
+                "Message: ${e.toString()}",
+            userErrorMessage: "Beim Verarbeiten des eingescannten Codes ist ein"
+                "Fehler aufgetreten. Fehlercode: base32TotpSecretInvalid");
+      }
+
+      final cardDetails = CardDetails(fullName, randomBytes, unixExpirationDate,
+          cardType, regionId, base32TotpSecret);
+      _cardDetailsModel.setCardDetails(cardDetails);
+    } on Exception catch (e) {
+      return QRCodeParseResult(
+          hasError: true,
+          internalErrorMessage: e.toString(),
+          userErrorMessage: "Ein unerwarteter Fehler ist aufgetreten.");
+    }
```

Comment:
I do not like the C-Style error handling here. It bloats up the code and makes it hard to understand what's really happening here. Why not just throw exceptions of one or multiple custom exception classes?

Reply:
I'd like to refactor that later after the merge. Unless you want to do it now

Reply:
No, thats okay 👍🏽